### PR TITLE
[11.0] Create module website_event_questions_open_ended

### PR DESCRIPTION
**Pull request by grindtildeath**

### CODE REVIEW

File: website_event_questions_open_ended/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from . import event
```
Comment (ghost) [inline website_event_questions_open_ended/models/__init__.py]: newline

**Review by tschanzt — APPROVED** [review]

File: website_event_questions_open_ended/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    'author': 'Camptocamp, Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'category': 'Others',
+    'depends': [
+        'website_event_questions',
+    ],
+    'website': 'http://www.camptocamp.com',
+    'data': [
+        'security/ir.model.access.csv',
+        'views/event_views.xml'
```
Comment (ghost) [inline website_event_questions_open_ended/__manifest__.py]: comma

File: website_event_questions_open_ended/models/event.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            vals.pop('answer_ids')
+        return super().create(vals)
+
+
+class EventAnswerOpen(models.Model):
+
+    _name = 'event.answer.open'
+
+    sequence = fields.Integer(default=10)
+    question_id = fields.Many2one('event.question', required=True,
```
Comment (ghost) [inline website_event_questions_open_ended/models/event.py]: would be nice to have those params on a dedicated line each :)

File: website_event_questions_open_ended/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    ],
+    'installable': True,
```
Comment (ghost) [inline website_event_questions_open_ended/__manifest__.py]: default?

**Review by ghost — APPROVED** [review]
LGTM w/ some nitpicking remarks :slightly_smiling_face:

File: website_event_questions_open_ended/security/ir.model.access.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+event_answer_open_all,event.answer.open.all,model_event_answer_open,,1,1,1,0
```
Comment (simahawk) [inline website_event_questions_open_ended/security/ir.model.access.csv:2]: no one can unlink?
You should have some event groups

**Review by simahawk — CHANGES_REQUESTED** [review]

File: website_event_questions_open_ended/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+from odoo.addons.website_event_questions.controllers.main import WebsiteEvent
+
+
+class WebsiteEventController(WebsiteEvent):
+
+    def _process_registration_details(self, details):
+        ''' Process data posted from the attendee details form. '''
+        # FIXME the call to super here goes directly to website_event's controller skipping website_event_questions' controller  # noqa
+        registrations = super(
+            WebsiteEvent, self)._process_registration_details(details)
```
Comment (simahawk) [inline website_event_questions_open_ended/controllers/main.py]: due to the black magic inheritance of odoo controllers: when you inherit them too much you have such issues.
Try this:
```python
question_controller = WebsiteEvent()
registrations = question_controller._process_registration_details(details)
```

**Review by simahawk — CHANGES_REQUESTED** [review]

File: website_event_questions_open_ended/controllers/main.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        for registration in registrations:
+            open_answers = []
+            for key, value in registration.items():
+                if key.startswith('open-answers-'):
+                    open_answers.append([0, 0, {
+                        'question_id': key[13:],
```
Comment (simahawk) [inline website_event_questions_open_ended/controllers/main.py]: :open_mouth: no better way to slice this key?
`key.split('open-answers-')[-1]`?

**Review by simahawk — CHANGES_REQUESTED** [review]

**Review by simahawk — APPROVED** [review]

**Pull request merged**